src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/openstack/OpenstackJCloudsProvider.java

```diff
@@ -93,12 +99,21 @@
 
     }
 
-    private CreateServerOptions createOptions(Instance instance) {
+    private CreateServerOptions createOptions(Infrastructure infrastructure, Instance instance) {
 
-        CreateServerOptions createServerOptions = new CreateServerOptions().keyPairName(instance.getCredentials()
-                                                                                                .getPublicKeyName())
-                                                                           .userData(buildScriptToExecuteString(instance.getInitScript()).getBytes());
+        // Acquire or generate KeyPair name
+        String publicKeyName;
+        if ((instance.getCredentials() == null) || (instance.getCredentials().getPublicKeyName() == null) ||
+            (instance.getCredentials().getPublicKeyName().equals(""))) {
```

Comment:
I think it is more clear to use `.isEmpty()` than `.equals("")`

Reply:
Done